src/main/java/org/mastodon/mamut/fitting/FitEllipsoidPlugin.java

```diff
@@ -195,131 +293,141 @@ private < T extends RealType< T > > void process( final SourceAndConverter< T >
-			final AffineTransform3D zeroMinSourceToGlobal = sourceToGlobal.copy();
-			final AffineTransform3D shiftToMin = new AffineTransform3D();
-			shiftToMin.translate( lMin[ 0 ], lMin[ 1 ], lMin[ 2 ] );
-			zeroMinSourceToGlobal.concatenate( shiftToMin );
-			final ArrayList< Edgel > gEdgels = Edgels.transformEdgels( lEdgels, zeroMinSourceToGlobal );
-			final ArrayList< Edgel > filteredEdgels = Edgels.filterEdgelsByOcclusion( Edgels.filterEdgelsByDirection( gEdgels, gCenter ), gCenter, maxAngle, maxFactor );
-
-			EdgelsOverlay edgelsOverlay;
-			if ( DEBUG )
-			{
-				edgelsOverlay = new EdgelsOverlay( filteredEdgels, 0.01 );
-				BdvFunctions.showOverlay( edgelsOverlay, "filtered edgels", Bdv.options().addTo( bdv ) );
-			}
+		final double[] centerInGlobalCoordinates = spot.positionAsDoubleArray();
+		final ArrayList< Edgel > filteredEdgels = Edgels.filterEdgelsByOcclusion(
+				Edgels.filterEdgelsByDirection( gEdgels, centerInGlobalCoordinates ), centerInGlobalCoordinates,
+				maxAngle, maxFactor );
 
-			final long t1 = System.currentTimeMillis();
-			final Ellipsoid ellipsoid = SampleEllipsoidEdgel.sample(
-					filteredEdgels,
-					gCenter,
-					numSamples,
-					outsideCutoffDistance,
-					insideCutoffDistance,
-					angleCutoffDistance,
-					maxCenterDistance );
-			if ( ellipsoid == null )
-				return;
-			final long t2 = System.currentTimeMillis();
-//			System.out.println( t2 - t1 );
-
-			if ( DEBUG )
-			{
-				BdvFunctions.showOverlay( new EllipsoidOverlay( ellipsoid ), "fitted ellipsoid", Bdv.options().addTo( bdv ) );
-				final Map< Edgel, Double > costs = SampleEllipsoidEdgel.getCosts(
-						filteredEdgels,
-						ellipsoid,
-						outsideCutoffDistance,
-						insideCutoffDistance,
-						angleCutoffDistance );
-				edgelsOverlay.setCosts( costs );
-			}
+		final Ellipsoid ellipsoid = SampleEllipsoidEdgel.sample(
+				filteredEdgels,
+				centerInGlobalCoordinates,
+				numSamples,
+				numCandidates,
+				outsideCutoffDistance,
+				insideCutoffDistance,
+				angleCutoffDistance,
+				maxCenterDistance );
+
+		if ( DEBUG )
+			showBdvDebugWindow( source, appModel, outsideCutoffDistance, insideCutoffDistance, angleCutoffDistance, sourceToGlobal, input, filteredEdgels, ellipsoid );
+		return ellipsoid;
+	}
+
+	private static RandomAccessibleInterval< ? > cropSpot( AffineTransform3D sourceToGlobal, RandomAccessibleInterval< ? > frame, Spot spot )
+	{
+		final double[] centerInGlobalCoordinates = spot.positionAsDoubleArray();
+		final double radius = Math.sqrt( spot.getBoundingSphereRadiusSquared() );
+		final double[] centerInLocalCoordinates = new double[ 3 ];
+		sourceToGlobal.applyInverse( centerInLocalCoordinates, centerInGlobalCoordinates );
 
-			spot.setPosition( ellipsoid );
-			spot.setCovariance( ellipsoid.getCovariance() );
+		final double[] scale = extractScale( sourceToGlobal );
+
+		final long[] lMin = new long[ 3 ];
+		final long[] lMax = new long[ 3 ];
+
+		for ( int d = 0; d < 3; ++d )
```

Comment:
```suggestion
		for ( int d = 0; d < 3; d++ )
```
Post increment is more usual in for loops and thus easier to read?

Reply:
I agree. But lets do this in a different PR, that fixes all the pre increments in the for loops.